### PR TITLE
[WIP] build: update everything to target-lexicon 0.9

### PR DESCRIPTION
**Pull request by m4b**
ref #1197

### CODE REVIEW

File: cranelift-object/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ edition = "2018"
 [dependencies]
 cranelift-module = { path = "../cranelift-module", version = "0.48.0" }
 object = { version = "0.15.0", default-features = false, features = ["write"] }
-target-lexicon = "0.8.1"
+target-lexicon = "0.8"
```
Comment (bnjbvr) [inline cranelift-object/Cargo.toml]: nit: 0.9

**Comment by bnjbvr** [comment]
Thanks for this PR! I didn't know `object` was also using this; thanks for updating `target-lexicon` there. Should we try to have a new version of `object` released, or just point our dependency on `object` to the github's commit in the meanwhile?

**Comment by philipc** [comment]
I can do an `object` release when needed, but from what I can tell there will be a new breaking version of target-lexicon soon (https://github.com/CraneStation/target-lexicon/pull/35), so I'd rather not churn out unneeded releases if you don't need to publish this yet.

**Comment by bnjbvr** [comment]
Since the new version of `faerie` has been published as a patch version, Cargo can pick it up when collecting dependencies for the same minor. So every time we ask for faerie 0.x.0, it will pick up the latest patch version. This causes build issues on master, because now we have conflicting versions of `target-lexicon` dependencies in the dependency tree. We need this PR.

**Comment by bnjbvr** [comment]
Closing this one, in favor of #1206. Thanks!

**Comment by m4b** [comment]
Derp sorry about that ! I can yank the minor and just republish 0.12 if it matters anymore

**Comment by bnjbvr** [comment]
No worries. Yeah it might be nice for other people using `faerie` to remove the patch version and publish a 0.12 instead; for what it's worth, updating in Cranelift would then just work.

**Comment by m4b** [comment]
Done!

**Comment by m4b** [comment]
Actually now you’re broken again because you’re depending on yanked version without lock file :(

**Comment by m4b** [comment]
I have unyanked this to unbreak you in the meantime; let me know once you've transitioned (no rush), and i'll yank the minor patch again... really sorry about this :( :face_with_head_bandage:

**Comment by bnjbvr** [comment]
(Wanted to answer earlier but oh well) No worries, this happens! Thanks for maintaining the faerie crate in the first place :)

**Comment by m4b** [comment]
No worries :) I saw 0.12 get added and then I yanked. Also 0.13 has been published which removed failure thanks to @alexcrichton

Also thank you for working on cranelift and friends ! :)

**Pull request closed without merging**